### PR TITLE
feature: forecaster hyperparams and eval metrics

### PR DESCRIPTION
**Pull request by egordm**
This pull request introduces significant improvements to probabilistic forecasting in the OpenSTEF codebase, focusing on evaluation metrics, input validation, and model configuration. The main highlights are the implementation of a vectorized mean pinball loss metric, integration of evaluation metric selection into model hyperparameters, enhanced input validation, and updates to XGBoost and GBLinear forecaster defaults. These changes improve model reliability, flexibility, and consistency across the codebase.

**Evaluation metric improvements**
* Implemented a vectorized `mean_pinball_loss` function in `metrics_probabilistic.py`, enabling efficient multi-quantile evaluation and matching scikit-learn's per-quantile average. This function is now used as the default evaluation metric for quantile regression models. [[1]](diffhunk://#diff-29333106e1578bab331564b02e7e19ce72a0c40456df77ba05aa2dc1ed94cf5cR218-R272) [[2]](diffhunk://#diff-e2d9a4f3945ad1d4618260765e2b61dbe45215630230c46ba4fb3979fe65c151R157-R186)
* Added a new utility module `evaluation_functions.py` with `EvaluationFunctionType` and `get_evaluation_function`, allowing models to easily select and use evaluation metrics based on configuration.

**Model configuration and integration**
* Updated both GBLinear and XGBoost forecasters to support configurable evaluation metrics via the new `evaluation_metric` hyperparameter, defaulting to `mean_pinball_loss`. Objective and evaluation functions are now set using dedicated utility functions for greater flexibility. [[1]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L25-R33) [[2]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L55-R74) [[3]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5R203-R212) [[4]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3L25-R29) [[5]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3R98-R102) [[6]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3L300-R313)
* Adjusted default hyperparameters for GBLinear (`objective`, `reg_lambda`) and XGBoost (`n_estimators`, `random_state`, `early_stopping_rounds`) to improve out-of-the-box performance and reproducibility. [[1]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L55-R74) [[2]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3L64-R68) [[3]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3L152-R164)

**Input validation and reliability**
* Added input validation to GBLinear forecaster's `fit` and `predict` methods, raising a new `InputValidationError` if NaN values are present in input data. The exception class was introduced in `openstef_core/exceptions.py`. [[1]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L25-R33) [[2]](diffhunk://#diff-58711f6d17c74c97b50c4378ed26da555f7648a6151b7c983452d4cdf568bd11R107-R110) [[3]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L237-R255) [[4]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5R283-R291)
* Ensured prediction outputs from both forecasters are reshaped to match the expected quantile dimensions, improving consistency and downstream compatibility. [[1]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5R283-R291) [[2]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3L375-R388)

**Testing and validation**
* Added a unit test to verify that the new `mean_pinball_loss` implementation matches the average of scikit-learn's per-quantile pinball loss, ensuring correctness and reliability.

**Codebase maintenance**
* Refactored imports and removed unused code for clarity and maintainability in multiple modules. [[1]](diffhunk://#diff-1725b6c0037fdfb4a0cffae2fcdfa37f98516045a969b09bf8f29b909f2fd9a5L14) [[2]](diffhunk://#diff-a933e569b6f94894ab18b8d097fdf9804d98dbfb372019d35ee3781624c165a3L12) [[3]](diffhunk://#diff-4b74bf8fb91af0e8d92f246a3a6ba11fab16e2e84a0753c65780109288efa5b6L12-R14)

### CODE REVIEW

File: packages/openstef-beam/tests/unit/metrics/test_metrics_probabilistic.py [hunk 1]
```diff
@@ -151,3 +154,33 @@ def test_mean_absolute_calibration_error() -> None:
...
+        [
+            y_true + rng.normal(0, 0.7, size=n) - 0.4,  # q=0.1
+            y_true + rng.normal(0, 0.5, size=n) + 0.0,  # q=0.5
+            y_true + rng.normal(0, 0.7, size=n) + 0.4,  # q=0.9
+        ],
+        axis=1,
+    )
+
+    # Act
+    ours = mean_pinball_loss(y_true=y_true, y_pred=y_pred, quantiles=quantiles)
```
Comment (MvLieshout) [inline packages/openstef-beam/tests/unit/metrics/test_metrics_probabilistic.py]: "ours" feels a bit weird here although I understand where it comes from given we want to compare to sklearn.

Maybe `result_openstef` would be nicer?

Comment (egordm) [reply packages/openstef-beam/tests/unit/metrics/test_metrics_probabilistic.py]: Fixed.
